Mutagen.Bethesda.Analyzers.Skyrim/Record/Location/MissingFieldsAnalyzer.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Analyzers.SDK.Results;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Plugins;
+using Mutagen.Bethesda.Skyrim;
+using Noggog;
+namespace Mutagen.Bethesda.Analyzers.Skyrim.Record.Location;
+
+public class MissingFieldsAnalyzer : IIsolatedRecordAnalyzer<ILocationGetter>
+{
+    public static readonly TopicDefinition NoParentLocation = MutagenTopicBuilder.DevelopmentTopic(
+            "No Parent Location",
+            Severity.Suggestion)
+        .WithoutFormatting("Location has no parent location - this is likely a mistake - only top level locations should have no parent location");
+
+    public IEnumerable<TopicDefinition> Topics { get; } = [NoParentLocation];
+
+    private static readonly HashSet<FormKey> ValidTopLevelLocations =
```

Comment:
Yeah I forgot to integrate the valid top level locations list.
This is pretty much just arbitrary. There are some valid ones like the top level location for the Skyrim worldspace (could maybe be found by removing any locations that are assigned to locations, I'll add that). Some are sort of hardcoded or some are part of daedric realms which don't really belong to any bigger location. In the end everything in the real world will be part of tamriel or nirn so that should always have a parent location. There are also some technical locations like the `PersistAll` which should always be ignored.
I'm not super happy with this whitelist approach here, and people would probably would like to be able to configure that stuff on their own potentially. I just included the most universal ones here for now. There might still be some false positives from the analyzer though - that'd be up to the user to decide currently.